lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,677 @@
+        self.profile_name = None
+        self.origins = None
+        self.tags = None
+        self.origin_host_header = None
+        self.origin_path = None
+        self.content_types_to_compress = None
+        self.is_compression_enabled = None
+        self.is_http_allowed = None
+        self.is_https_allowed = None
+        self.query_string_caching_behavior = None
+
+        self.cdn_client = None
+
+        self.results = dict(changed=False)
+
+        super(AzureRMCdnendpoint, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                 supports_check_mode=True,
+                                                 supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.cdn_client = self.get_cdn_client()
+
+        to_be_updated = False
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        if self.query_string_caching_behavior:
+            self.query_string_caching_behavior = _snake_to_camel(self.query_string_caching_behavior)
+
+        response = self.get_cdnendpoint()
+
+        if self.state == 'present':
+
+            if not response:
+
+                if self.started is None:
+                    # If endpoint dosen't exist and no start/stop operation specified, create endpoint.
+                    if self.origins is None:
+                        self.fail("Origins is not provided when trying to create endpoint")
+                    self.log("Need to create the Azure CDN endpoint")
+
+                    if not self.check_mode:
+                        result = self.create_cdnendpoint()
+                        self.results['id'] = result['id']
+                        self.results['host_name'] = result['host_name']
+                        self.log("Creation done")
+
+                    self.results['changed'] = True
+                    return self.results
+
+                else:
+                    # Fail the module when user try to start/stop a non-existed endpoint
+                    self.log("Can't stop/stop a non-existed endpoint")
+                    self.fail("This endpoint is not found, stop/start is forbidden")
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+
+                    if self.started is False and response['resource_state'] == 'Running':
+                        self.log("Need to stop the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.stop_cdnendpoint()
+                            self.results['id'] = result['id']
```

Comment:
Don't we always have the ID and hostname from the original `get_cdnendpoint` response? Why is this repeated everywhere?

Reply:
fixed